### ISSUE TITLE
Crash on Android 2.3.4 - java.lang.StackOverflowError on Style B view init

### ISSUE THREAD

**Issue by barbeau** [OneBusAway/onebusaway-android#350]
For `develop` branch, on an HTC EVO Shift w/ Android 2.3.4.

When tapping on a bus stop and viewing arrivals sorted by Route in Style B, the app crashes and I see:

~~~
FATAL EXCEPTION: main
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime: java.lang.StackOverflowError
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.text.Styled.measureText(Styled.java:430)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.text.Layout.measureText(Layout.java:2065)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.text.Layout.getLineMax(Layout.java:1020)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.text.Layout.draw(Layout.java:612)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.widget.TextView.onDraw(TextView.java:4328)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.View.draw(View.java:6970)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1732)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1730)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1730)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1730)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1730)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.View.draw(View.java:6973)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.widget.FrameLayout.draw(FrameLayout.java:357)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1732)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.View.draw(View.java:6973)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1732)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.widget.AbsListView.dispatchDraw(AbsListView.java:1480)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.widget.ListView.dispatchDraw(ListView.java:3186)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.View.draw(View.java:7082)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.widget.AbsListView.draw(AbsListView.java:2588)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1732)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1730)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1730)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1730)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1730)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.View.draw(View.java:6973)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1732)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at com.sothree.slidinguppanel.SlidingUpPanelLayout.drawChild(SlidingUpPanelLayout.java:1200)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.View.draw(View.java:6973)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at com.sothree.slidinguppanel.SlidingUpPanelLayout.draw(SlidingUpPanelLayout.java:1243)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1732)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1730)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.support.v4.widget.DrawerLayout.drawChild(DrawerLayout.java:1089)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1730)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.View.draw(View.java:6973)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.widget.FrameLayout.draw(FrameLayout.java:357)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1732)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.View.draw(View.java:6973)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.support.v7.internal.widget.ActionBarOverlayLayout.draw(ActionBarOverlayLayout.java:500)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1732)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1730)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.drawChild(ViewGroup.java:1730)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewGroup.dispatchDraw(ViewGroup.java:1459)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.View.draw(View.java:6973)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.widget.FrameLayout.draw(FrameLayout.java:357)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at com.android.internal.policy.impl.PhoneWindow$DecorView.draw(PhoneWindow.java:1949)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewRoot.draw(ViewRoot.java:1600)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime:     at android.view.ViewRoot.performTraversals(ViewRoot.java:1321)
10-30 17:33:41.509 6527-6527/? E/AndroidRuntime: 	at android.view.ViewRoot.handleMessage(ViewRoot.java:195
~~~

Looks like this may be a repeat of #275.

**Comment by barbeau** [comment]
I believe the only major additions to the Style B layout following the fix for #275 are the star icon, status labels in the ETA rows, and the real-time indicator.

At this point I think it's going to be impossible to try and support Arrival Style B (sort by route) back to Gingerbread.  There are just too many views (for Gingerbread), and the thread stack size for Gingerbread isn't large enough to support them.  We already tried to trim the view hierarchy down in the fix for #275, and we're back at the same problem again.

The simplest solution is just to lock arrivals to be sorted by Style A ETA (the same as in OBA Android v1.x) on Gingerbread, and do not offer the menu option to change ETA sorting.

**Comment by barbeau** [comment]
Fixed in `develop` via https://github.com/OneBusAway/onebusaway-android/commit/8d1807c4e023d218be828febaf5a87d995f40192.
